### ISSUE TITLE
Server list for global search

### ISSUE THREAD

**Issue by Thatoo** [matrix-org/synapse#16245]
Would it be possible to list in homeserver.yaml file a list of servers that could serve for users to make a global search when they are looking for new rooms?

**Comment by clokep** [comment]
I'm assuming you're asking about the public room directory, please correct me if I'm wrong.

The client provides the server for which to search (defaulting to the local server). It currently isn't possible to search multiple servers at once.

**Comment by Thatoo** [comment]
Yes, you're correct and yes the request is to be able to search multiple servers at once : a list of server written in the client .well-known and  those added by the user.

**Comment by clokep** [comment]
I see this as two pieces:

1. By default, instead of performing only a local room directory search, include additional administrator configured servers. It isn't clear to me that the spec allows for this, but it doesn't seem too off base either.
2. Allow clients to provide multiple servers to the room directory search API. This would require a spec change.
